src/stream/src/from_proto/chain.rs

```diff
@@ -43,26 +50,98 @@ impl ExecutorBuilder for ChainExecutorBuilder {
         // its schema.
         let schema = snapshot.schema().clone();
 
-        if node.disable_rearrange {
-            let executor = ChainExecutor::new(
+        let executor = match node.chain_type() {
+            ChainType::Chain => ChainExecutor::new(
                 snapshot,
                 mview,
                 upstream_indices,
                 progress,
                 schema,
                 params.pk_indices,
-            );
-            Ok(executor.boxed())
-        } else {
-            let executor = RearrangedChainExecutor::new(
+            )
+            .boxed(),
+            ChainType::Rearrange => RearrangedChainExecutor::new(
```

Comment:
Just to confirm: Will `ChainExecutor` or `RearrangedChainExecutor` be used in any cases?

Reply:
`ChainExecutor ` is still used in `StreamIndexScan` as before, while `RearrangedChainExecutor ` will be replaced by `BackfillExecutor`.